### PR TITLE
Check current draw in intake

### PR DESCRIPTION
**Pull request by brendanburkhart**

### CODE REVIEW

**Comment by calebeby** [comment]
Good improvement, but can you change the logic here?

```
if has bunny
  if button pressed
    spit bunny
else
  intake bunny
```

**Comment by brendanburkhart** [comment]
How do you want to tell if there is a bunny? You want it to constantly have the intake running if it doesn't have a bunny?

**Comment by calebeby** [comment]
Current draw. Once it spikes, set has_bunny to true. Once you spit it, set it to false

**Comment by calebeby** [comment]
Also even when it has a bunny you should run it a little bit (intake) so that the bunny does not have a chance of falling out

**Comment by calebeby** [comment]
No this doesn't need to be a magicbot state machine... We don't need timed states, just spit while the button is pressed

**Comment by brendanburkhart** [comment]
You don't think the fsm is cleaner and less complex than a mess of if-else all cramped into `execute`? I agree though, the timed state is unnecessary.

**Comment by calebeby** [comment]
The if/else will not be bad

**Comment by brendanburkhart** [comment]
Ok yeah the if-else is actually a **lot** better

**Comment by calebeby** [comment]
The current won't be spiking when the intake is running to hold the bunny in

**Comment by brendanburkhart** [comment]
It should spike at little, but you're right, probably not enough to differentiate it from just intaking the bunny. We can either try to define and tune two thresholds, one for intaking and one for holding, or use a state variable.

**Comment by calebeby** [comment]
State variable

**Comment by brendanburkhart** [comment]
The bunny isn't going to get ejected, it'll be pulled back up too quickly.

**Comment by calebeby** [comment]
No the driver will hold on the outake button

File: components/intake.py
Comment (calebeby) [inline components/intake.py]: Uhhh... why camel case?

File: components/intake.py
Comment (calebeby) [inline components/intake.py]: Is `has_bunny` a better name?

**Review by calebeby — CHANGES_REQUESTED** [review]

File: components/intake.py
Comment (brendanburkhart) [reply components/intake.py]: It should using an underscore, but I don't really have a preference between `holding_bunny` and `has_bunny`, whatever you want.

**Comment by brendanburkhart** [comment]
I think they're going to have to hold it for a decent amount of time once it's been jammed up, should I remap the release to something easier to access like the trigger or a button on the top?

File: components/intake.py
Comment (brendanburkhart) [reply components/intake.py]: Shh.... it never happened...

**Comment by calebeby** [comment]
I think it will be less than a second to eject. And no, I like the button that it's on, until we move it to an operator joystick. I'd like to reserve the trigger for some cool driving trick, like rotate 180deg or something useful

File: components/intake.py
Comment (calebeby) [reply components/intake.py]: I prefer has_bunny

File: components/intake.py
Comment (calebeby) [inline components/intake.py]: intaken?

Comment (brendanburkhart) [reply components/intake.py]: The line was too long, I shortened it. Should it be intook or something? I'm think intaken is the correct tense...

Comment (calebeby) [reply components/intake.py]: Could we inline this function?

Comment (brendanburkhart) [reply components/intake.py]: Yeah. Generally inlining into an if-else tree just harms readability though. Do you want me to do it?
```
if self.holding_bunny:
            if self.release_bunny:
                self.intake_motor.set(-0.6)
                self.holding_bunny = False
            else:
                self.intake_motor.set(0.1)
        else:
            self.intake_motor.set(0.5)
            if self.picked_up_bunny():
                self.holding_bunny = True
```
vs
```
if self.holding_bunny:
            if self.release_bunny:
                self.intake_motor.set(-0.6)
                self.holding_bunny = False
            else:
                self.intake_motor.set(0.1)
        else:
            self.intake_motor.set(0.5)
            if self.pdp.getCurrent(self.intake_pdp_channel) > self.max_current:
                self.holding_bunny = True
```

Comment (calebeby) [reply components/intake.py]: Yeah I think so

Comment (brendanburkhart) [reply components/intake.py]: Sigh... it just hides the purpose with the implementation for the sake of 4 LOC.

**Review by calebeby — DISMISSED** [review]
OK, lgtm once you inline that function

Let's not merge this till we have tested it on the robot though

File: components/intake.py
Comment (calebeby) [reply components/intake.py]: OK, you don't have to, but I just think it is more clear if it is inlined. Your choice

File: components/intake.py
Comment (calebeby) [inline components/intake.py]: Let's make this full speed. The bag motor is pretty slow and weak, let's drop the bunny as fast as possible

Comment (brendanburkhart) [reply components/intake.py]: Just that one or the others as well?

Comment (brendanburkhart) [reply components/intake.py]: I’ll increase them all for now, we can tune it tomorrow.

**Review by calebeby — APPROVED** [review]

**Pull request merged**
